docs/tutorial.py

```diff
@@ -0,0 +1,95 @@
+# ruff: noqa
+"""
+The kasa library is fully async and methods that perform IO need to be run inside an async couroutine.\n
+These examples assume you are following the tutorial inside `asyncio REPL` (python -m asyncio) or the code
+is running inside an async function (`async def`).
+
+
+The main entry point for the api is :meth:`~kasa.Discover.discover` and
+:meth:`~kasa.Discover.discover_single` which return Device objects.\n
+Most newer devices require your tplink cloud username and password but this can be ommitted for older devices.
+
+>>> from kasa import Device, Discover, Credentials
+
+:func:`~kasa.Discover.discover` returns a list of devices on your network
+
+>>> devices = await Discover.discover(credentials=Credentials("user@mail.com", "great_password"))
+>>> for dev in devices:
+>>>     await dev.update()
+>>>     print(dev.host)
+127.0.0.1
+127.0.0.2
+
+:meth:`~kasa.Discover.discover_single` returns a single device by hostname
+
+>>> dev = await Discover.discover_single("127.0.0.1", credentials=Credentials("user@mail.com", "great_password"))
+>>> await dev.update()
+>>> dev.alias
+Living Room
+>>> dev.model
+L530
+>>> dev.rssi
+-52
+>>> dev.mac
+5C:E9:31:00:00:00
+
+You can update devices by calling set methods. You need to call :meth:`~kasa.Device.update()` for updates to be propogated
+back to the device.
+
+>>> await dev.set_alias("Dining Room")
+>>> await dev.update()
+>>> dev.alias
+Dining Room
+
+Different groups of functionality are supported by modules which you can access via :attr:`~kasa.Device.modules` with a typed
+key from :class:`~kasa.Module`.\n
+Modules will only be available on the device if they are supported but some individual features of
+a module may not be available for your device.  You can check for this with `is_feature`, e.g. is_color.
```

Comment:
```suggestion
Modules will only be available on the device if they are supported but some individual features of a module may not be available for your device.
You can check the availability using ``is_``-prefixed properties like `is_color`.
```
It's generally preferred to make a new line after every sentence, as this makes diffs smaller when something gets modified.